rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java

```diff
@@ -93,4 +93,77 @@ void removeSimpleRepoWithUrlOnly() {
           )
         );
     }
+
+
+    @Test
+    void removePluginSimpleRepoWithIdAndUrl() {
+
+        rewriteRun(
+          spec -> spec.recipe(new RemoveRepository("myRepo", "http://myrepo.maven.com/repo", null, null,
+            null, null, null,
+            null, null, null)),
+          pomXml(
+            """
+              <project>
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                <pluginRepositories>
+                  <pluginRepository>
+                    <id>myRepo</id>
+                    <url>http://myrepo.maven.com/repo</url>
+                  </pluginRepository>
+                  <pluginRepository>
+                    <id>identicalRepo</id>
+                    <url>http://myrepo.maven.com/repo</url>
+                  </pluginRepository>
+                </pluginRepositories>
```

Comment:
I maintained the same behavior, but I don't agree with the existing functionality

In this case, the user provided the repo id `myRepo` but this recipe removes also the other repo with id `identicalRepo`

Even if it is the same URL, I don't think that this recipe should do that

Reply:
@yeikel I agree. It feels like a bug that the recipe would remove a repository with an ID other than the one the user specified.

Reply:
I'll fix it before it is merged. Thank you for your review

Reply:
@sambsnyd I updated it with 8dc10b861b38c20b6a0711fb49daff416cfd0691 to only remove if both match

One additional detail that I noticed about this recipe is that it contains additional keys that are not necessarily used in the recipe

![image](https://user-images.githubusercontent.com/8935151/216793237-389c178a-e0b6-46fd-8026-1655c93e769b.png)


I tentatively pushed 379aa19194cdde9f6398d2a85d6876b81c811dd0 to remove them and simplify them, but we can revert it if you think that it is needed.

Reply:
Thanks for cleaning up the options as well!